rules/language/python.yara

```diff
@@ -0,0 +1,18 @@
+rule dangerous_fstring : notable {
```

Comment:
Test file:
```py
import os

dangerous = f'{exec(os.system("rm -rf /"))}'
dangerous2 = f"{eval('os.system("rm -rf /"))')}"
dangerous3 = f'''{os.system("rm -rf /")}'''
dangerous4 = f"""{os.system("rm -rf /")}"""
```

Output:
```
❯ go run . ~/Downloads/test_dangerous.py
/Users/egibs/Downloads/test_dangerous.py [⚠️ MEDIUM]
-------------------------------------------------------------------------------------------------------------------
RISK  KEY                      DESCRIPTION                                     EVIDENCE
-------------------------------------------------------------------------------------------------------------------
MED   exec/program             executes external programs                      system(
MED   fs/file/delete/forcibly  Forcibly recursively deletes files using rm -R  rm -rf /
MED   language/python          detect comments up to PEP8's limit and ignore   f"""{os.system("rm -rf /")}
                                                                               f"{eval('os.system("rm -rf /"))')}
                                                                               f'''{os.system("rm -rf /")}
                                                                               f'{exec(os.system("rm -rf /"))}
MED   techniques/code_eval     detect comments up to PEP8's limit and ignore   f"""{os.system("rm -rf /")}
                                                                               f"{eval('os.system("rm -rf /"))')}
                                                                               f'''{os.system("rm -rf /")}
                                                                               f'{exec(os.system("rm -rf /"))}
-------------------------------------------------------------------------------------------------------------------
```

Original file with the same rules:
```
❯ go run . ~/Downloads/utilpep613.py
/Users/egibs/Downloads/utilpep613.py [✅ LOW]
----------------------------------------------------------------------
RISK  KEY               DESCRIPTION                     EVIDENCE
----------------------------------------------------------------------
LOW   ref/path/usr/bin  path reference within /usr/bin  /usr/bin/env
----------------------------------------------------------------------
```

Reply:
The downside here is that including the `python.yara` file evaluates the `dangerous_fstring` rule twice.

The way around this is to have a duplicate, `private` version of the rule in `code_eval.yara` to avoid reporting the matches but that's also redundant so this is an interesting problem.

@tstromberg any thoughts?

Reply:
I consolidated the new Python rules into `code_eval.yara` for now in [`c177ca4` (#168)](https://github.com/chainguard-dev/bincapz/pull/168/commits/c177ca42afe1c339031f61c55e8f199f565748cb).

Reply:
I'm going to have to test this, but my first thought is that trying to attempt this natively in YARA (without a preprocessor) is going to suffer from accuracy problems and/or scalability problems in trying to apply it across all of our rules.

Reply:
Makes sense. I wasn't entirely sold on this approach since it didn't seem scalable at all.